drivers/sensor/st/lps22hh/lps22hh.c

```diff
@@ -252,21 +227,72 @@ static int lps22hh_init_chip(const struct device *dev)
 	return 0;
 }
 
-static int lps22hh_init(const struct device *dev)
+static int lps22hh_pm_action(const struct device *dev, enum pm_device_action action)
 {
-	if (lps22hh_init_chip(dev) < 0) {
-		LOG_DBG("Failed to initialize chip");
-		return -EIO;
+	const struct lps22hh_config *cfg = dev->config;
+	int ret = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_TURN_ON:
+		/* Full chip bring‑up */
+		ret = lps22hh_init_chip(dev);
+		if (ret != 0) {
+			LOG_ERR("%s: failed to initialise chip: %d", dev->name, ret);
+		}
+		break;
+
+	case PM_DEVICE_ACTION_RESUME:
+		ret = lps22hh_set_odr_raw(dev, cfg->odr);
+		if (ret != 0) {
+			LOG_ERR("%s: Failed to set odr %d", dev->name, cfg->odr);
+		}
+		break;
+
+	case PM_DEVICE_ACTION_SUSPEND:
+		/* set odr to 0 for power-down mode */
+		ret = lps22hh_set_odr_raw(dev, 0);
+		if (ret != 0) {
+			LOG_ERR("%s: Failed to reset odr", dev->name);
+		}
+		break;
+
+	case PM_DEVICE_ACTION_TURN_OFF:
+		break;
+
+	default:
+		ret = -ENOTSUP;
+		break;
 	}
 
+	return ret;
+}
+
+static int lps22hh_init(const struct device *dev)
+{
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+	const struct lps22hh_config *const cfg = dev->config;
+	struct lps22hh_data *data = dev->data;
+
+	if (cfg->i3c.bus != NULL) {
+		/*
+		 * Need to grab the pointer to the I3C device descriptor
+		 * before we can talk to the sensor.
+		 */
+		data->i3c_dev = i3c_device_find(cfg->i3c.bus, &cfg->i3c.dev_id);
+		if (data->i3c_dev == NULL) {
+			LOG_ERR("Cannot find I3C device descriptor");
+			return -ENODEV;
+		}
+	}
```

Comment:
I actually would squash the two commit together, as this has to be called only once, but it was called only once before the PM commit. So it belongs there. Pls squash the two together.

Reply:
Talking about the i3c grab commit and the PM commit